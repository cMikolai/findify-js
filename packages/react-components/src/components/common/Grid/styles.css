$gutter: 20px;

.root{
  font-size: 0;
  line-height: 0;
  width: 100%;
}

.column{
  display: inline-block;
  box-sizing: border-box;
  min-height: 1px;
  font-size: $font-size-medium;
  line-height: $line-height;
  vertical-align: top;
}

@supports(display: flex) {
  .root{
    display: flex;
    flex-direction: row;
    align-items: top;
    font-size: inherit;
    line-height: inherit;
    flex-wrap: wrap;
  }
  .column{
    display: block;
  }
}

@for $i from 1 to 12 {
  .column-$i {
    width: calc(100 / 12 * $i)%;
  }
}
