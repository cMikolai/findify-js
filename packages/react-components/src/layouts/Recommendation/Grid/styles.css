.title {
  padding: 0;
  margin: 0;
  padding-bottom: 18px;
  text-align: center; 
}
