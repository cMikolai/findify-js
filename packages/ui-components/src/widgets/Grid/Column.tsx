import * as React from 'react';
import * as cx from 'classnames';
import { createEagerFactory } from 'recompose';

const styles = require('./styles.css');

export interface Props {
  className?: string;
  columnClass?: string;
  columnStyle?: React.CSSProperties;
  children?: React.ReactChild;
}

const getClassName = (props: Props) =>
  cx(styles.column, props.className, props.columnClass);

export const Column = createEagerFactory((props: Props) => (
  <div className={getClassName(props)} style={props.columnStyle}>
    {props.children}
  </div>
));
