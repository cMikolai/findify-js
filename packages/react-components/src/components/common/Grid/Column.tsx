/**
 * @module components/common/Grid
 */

import React from 'react';
import cx from 'classnames';
import { setDisplayName } from 'recompose';

import styles from 'components/common/Grid/styles.css';

/** List of props that GridColumn accepts */
export interface IGridColumnProps {
  /** Custom className for column */
  className?: string;
  /** Column className */
  columnClass?: string;
  /** Column inline style */
  columnStyle?: React.CSSProperties;
  /** Contents of the column */
  children?: React.ReactChild;
}

/** Used to concatenate & build complete className from props */
const getClassName = (props: IGridColumnProps) =>
  cx(styles.column, props.className, props.columnClass);

const GridColumn = setDisplayName('GridColumn')((props: IGridColumnProps) => (
  <div className={getClassName(props)} style={props.columnStyle} role='listitem' tabIndex={0}>
    {props.children}
  </div>
));

export default GridColumn;
