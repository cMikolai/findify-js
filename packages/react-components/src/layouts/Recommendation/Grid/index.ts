import React from 'react';
import { connectConfig } from '@findify/react-connect';
import { compose, setDisplayName } from 'recompose';
import withTheme from 'helpers/withTheme';
import view from 'layouts/Recommendation/Grid/view';
import styles from 'layouts/Recommendation/Grid/styles.css';

export default compose(
  setDisplayName('GridRecommendation'),
  withTheme(styles),
  connectConfig
)(view);
