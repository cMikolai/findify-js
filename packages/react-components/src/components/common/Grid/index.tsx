import React from 'react';
import { compose, withPropsOnChange, setDisplayName, defaultProps } from 'recompose';
import { memoize } from 'lodash';
import cx from 'classnames';
import styles from './styles.css'

import { Column } from './Column';

type Props = {
  columns: string,
  style?: React.StyleHTMLAttributes<any>,
  children: any,
  theme?: { [className: string]: string }
}

type OwnProps = Props & {
  theme: { [className: string]: string }
}

const getClassName = memoize((columns, theme) =>
  columns.split('|').map(value => theme[`column-${value}`])
);

export const Grid = compose<OwnProps, Props>(
  setDisplayName('Grid'),
  defaultProps({ theme: styles }),
  withPropsOnChange(['columns', 'children'], ({ columns, children, theme }) => {
    const classNames = getClassName(columns, theme);
    return {
      children: React.Children.map(
        children,
        (child: React.ReactChild, index: number) =>
          child &&
          Column({
            key: child.key,
            children: child,
            className: classNames[index] || classNames[0],
            columnClass: child.props.columnClass,
            columnStyle: child.props.columnStyle,
          })
      ),
    };
  })
)(({ children, theme, style }) => (
  <div className={theme.root} style={style}>
    {children}
  </div>
));

export default Grid;
