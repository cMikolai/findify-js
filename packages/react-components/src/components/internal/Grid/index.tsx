import React from 'react';
import { compose, pure, withPropsOnChange, setDisplayName } from 'recompose';
import { memoize } from 'lodash';
import cx from 'classnames';
import styles from './styles.css'


import { Column } from './Column';



const getClassName = memoize(columns =>
  columns.split('|').map(value => styles[`column-${value}`])
);

export const Grid = compose(
  pure,
  setDisplayName('Grid'),
  withPropsOnChange(['columns', 'children'], ({ columns, children }: Props) => {
    const classNames = getClassName(columns);
    return {
      children: React.Children.map(
        children,
        (child: React.ReactChild, index: number) =>
          child &&
          Column({
            key: child.key,
            children: child,
            className: classNames[index] || classNames[0],
            columnClass: child.props.columnClass,
            columnStyle: child.props.columnStyle,
          })
      ),
    };
  })
)(({ children, className, style }) => (
  <div className={cx(styles.wrap, className)} style={style}>
    {children}
  </div>
));

export default Grid;
