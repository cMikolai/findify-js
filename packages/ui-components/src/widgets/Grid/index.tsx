import * as React from 'react';
import { compose, pure, withPropsOnChange, setDisplayName } from 'recompose';
import * as cx from 'classnames';
import { memoize } from 'lodash';

import { Column } from './Column';

const styles = require('./styles.css');

export interface Props {
  children?: React.ReactChild;
  columns: string;
  className?: string;
  style?: React.CSSProperties;
}

const getClassName = memoize(columns =>
  columns.split('|').map(value => styles[`column-${value}`])
);

export const Grid = compose<Props, Props>(
  pure,
  setDisplayName('Grid'),
  withPropsOnChange(['columns', 'children'], ({ columns, children }: Props) => {
    const classNames = getClassName(columns);
    return {
      children: React.Children.map(
        children,
        (child: React.ReactChild, index: number) =>
          child &&
          Column({
            key: child.key,
            children: child,
            className: classNames[index] || classNames[0],
            columnClass: child.props.columnClass,
            columnStyle: child.props.columnStyle,
          })
      ),
    };
  })
)(({ children, className, style }: Props) => (
  <div className={cx(styles.wrap, className)} style={style}>
    {children}
  </div>
));

export default Grid;
