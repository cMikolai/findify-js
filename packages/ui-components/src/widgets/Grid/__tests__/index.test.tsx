import * as React from 'react';
import { Column, Props as ColumnProps } from '../Column';
import Grid from '..';

describe('<Column />', () => {
  it('render correctly', () => {
    const properties: ColumnProps[] = [
      {},
      { className: 'foo' },
      { className: 'foo', columnClass: 'bar' },
      { columnClass: 'bar' },
      { columnStyle: { overflow: 'hidden' } },
      {
        className: 'foo',
        columnClass: 'bar',
        columnStyle: {
          padding: 2,
          margin: 14,
        },
      },
    ];
    properties.forEach((props: ColumnProps) => {
      const component = render(Column(props));
      expect(component).toMatchSnapshot();
    });
  });
});

describe('<Grid />', () => {
  describe('with only single column', () => {
    it('renders correctly', () => {
      const style = { margin: -12, color: '#ffa' };
      const component = render(
        <Grid columns="2" className="foo" style={style}>
          <div>foo</div>
          <div>bar</div>
        </Grid>
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe('with multiple columns separated by |', () => {
    it('renders correctly', () => {
      const cols = ['2|3', '4|5|6', '2|2|3|3|4'];
      cols.forEach((columns: string) => {
        const component = render(
          <Grid columns={columns} className="custom-wrapper">
            <div>foo</div>
            <div>bar</div>
            <div>baz</div>
          </Grid>
        );
        expect(component).toMatchSnapshot();
      });
    });
  });
});
