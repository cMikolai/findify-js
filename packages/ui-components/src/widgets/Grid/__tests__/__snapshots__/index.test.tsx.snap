// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> render correctly 1`] = `
<div
  class="column"
/>
`;

exports[`<Column /> render correctly 2`] = `
<div
  class="column foo"
/>
`;

exports[`<Column /> render correctly 3`] = `
<div
  class="column foo bar"
/>
`;

exports[`<Column /> render correctly 4`] = `
<div
  class="column bar"
/>
`;

exports[`<Column /> render correctly 5`] = `
<div
  class="column"
  style="overflow:hidden"
/>
`;

exports[`<Column /> render correctly 6`] = `
<div
  class="column foo bar"
  style="padding:2px;margin:14px"
/>
`;

exports[`<Grid /> with multiple columns separated by | renders correctly 1`] = `
<div
  class="wrap custom-wrapper"
>
  <div
    class="column column-2"
  >
    <div>
      foo
    </div>
  </div>
  <div
    class="column column-3"
  >
    <div>
      bar
    </div>
  </div>
  <div
    class="column column-2"
  >
    <div>
      baz
    </div>
  </div>
</div>
`;

exports[`<Grid /> with multiple columns separated by | renders correctly 2`] = `
<div
  class="wrap custom-wrapper"
>
  <div
    class="column column-4"
  >
    <div>
      foo
    </div>
  </div>
  <div
    class="column column-5"
  >
    <div>
      bar
    </div>
  </div>
  <div
    class="column column-6"
  >
    <div>
      baz
    </div>
  </div>
</div>
`;

exports[`<Grid /> with multiple columns separated by | renders correctly 3`] = `
<div
  class="wrap custom-wrapper"
>
  <div
    class="column column-2"
  >
    <div>
      foo
    </div>
  </div>
  <div
    class="column column-2"
  >
    <div>
      bar
    </div>
  </div>
  <div
    class="column column-3"
  >
    <div>
      baz
    </div>
  </div>
</div>
`;

exports[`<Grid /> with only single column renders correctly 1`] = `
<div
  class="wrap foo"
  style="margin:-12px;color:#ffa"
>
  <div
    class="column column-2"
  >
    <div>
      foo
    </div>
  </div>
  <div
    class="column column-2"
  >
    <div>
      bar
    </div>
  </div>
</div>
`;
